File: moviejava/app/src/main/java/com/example/moviejava/data/local/DatabaseModule.java
package com.example.moviejava.data.local;


import android.app.Application;

import androidx.room.Room;

import com.example.moviejava.data.local.dao.AppDao;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import dagger.hilt.InstallIn;
import dagger.hilt.components.SingletonComponent;

@Module
@InstallIn(SingletonComponent.class)

public class DatabaseModule {

    @Singleton
    @Provides
    public AppDatabase provideDatabase(Application application) {
        return Room.databaseBuilder(
                        application.getApplicationContext(),
                        AppDatabase.class,
                        "app_db.db"
                )
                .build();
    }

    @Provides
    @Singleton
    public AppDao provideAppDao(AppDatabase db) {
        return db.appDao();
    }
}

File: movie/app/src/main/java/com/example/movie/data/local/DatabaseModule.kt
package com.example.movie.data.local

import android.app.Application
import androidx.room.Room
import com.example.movie.data.local.dao.AppDao
import dagger.Module
import dagger.Provides
import dagger.hilt.InstallIn
import dagger.hilt.components.SingletonComponent
import javax.inject.Singleton

@Module
@InstallIn(SingletonComponent::class)

object DatabaseModule {

    @Singleton
    @Provides
    fun provideDatabase(application: Application): AppDatabase {
        return Room.databaseBuilder(
            application.applicationContext,
            AppDatabase::class.java,
            "app_db.db"
        )
            .allowMainThreadQueries().build()
    }

    @Provides
    @Singleton
    fun provideAppDao(db: AppDatabase): AppDao = db.appDao()
}